pageserver/src/tenant.rs

```diff
@@ -2647,36 +2647,46 @@ pub mod harness {
         }
     }
 
-    pub struct TenantHarness<'a> {
+    /// The harness provides a way to create mock certain tenant processes for testing.
+    /// It also creates a temporary dir that serves as a test repository, that gets removed on the harness drop.
+    pub struct TenantHarness {
+        // keep the struct to not to remove tmp dir during the test
+        temp_repo_dir: Option<TempDir>,
```

Comment:
I'm not sure I get it why it is an option (seems to compile fine without it) and do we really need the drop code? Dir will be cleaned on drop automatically

Reply:
Applied the patch, thanks.